packages/diffhtml/lib/tree/sync.js

```diff
@@ -71,7 +71,7 @@ export default function syncTree(oldTree, newTree, patches, parentTree, specialC
     // then splice it into the parent (if it exists) and run a sync.
     if (retVal && retVal !== newTree) {
       newTree.childNodes = [].concat(retVal);
-      syncTree(null, retVal, patches, newTree);
+      syncTree(oldTree !== empty ? oldTree : null, retVal, patches, newTree);
```

Comment:
Ah, that explains a lot, because it is generating SET_ATTRIBUTE patches for attributes that already match the DOM's current state. Will update with a comment.